### PR TITLE
[ETP-473] Tests/v2 snapshots Part 3

### PR DESCRIPTION
**Pull request by rafsuntaskin**
[ETP-473]

Adding Tests for 'Ticket/Item/Extra'

Adding more tests over the previous PR #1787

[ETP-473]: https://moderntribe.atlassian.net/browse/ETP-473

### CODE REVIEW

File: src/views/v2/tickets/item/extra/price.php [hunk 1]
```diff
@@ -14,11 +14,11 @@
  *
  * @version TBD
  *
+ * @var Tribe__Tickets__Ticket_Object      $post_id                     The Post object ID.
  * @var Tribe__Tickets__Ticket_Object      $ticket                      The ticket object.
  * @var Tribe__Tickets__Tickets            $provider                    The tickets provider class.
  * @var string                             $provider_id                 The tickets provider class name.
- * @var bool                               $show_original_price_on_sale True if it shows the original price on sale.
- * @var Tribe__Tickets__Commerce__Currency $currency
+ * @var Tribe__Tickets__Commerce__Currency $currency	                The Currency Object
```
Comment (tr1b0t) [inline src/views/v2/tickets/item/extra/price.php:21]: :no_entry_sign: Spaces must be used for mid-line alignment; tabs are not allowed

<sub>hash: dd1236ee81cefe1b9726d425adecffe2</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 1 error(s)

File: src/views/v2/tickets/item/extra/price.php [hunk 2]
```diff
@@ -14,11 +14,11 @@
  *
  * @version TBD
  *
- * @var Tribe__Tickets__Ticket_Object      $ticket                      The ticket object.
- * @var Tribe__Tickets__Tickets            $provider                    The tickets provider class.
- * @var string                             $provider_id                 The tickets provider class name.
- * @var bool                               $show_original_price_on_sale True if it shows the original price on sale.
- * @var Tribe__Tickets__Commerce__Currency $currency
+ * @var Tribe__Tickets__Ticket_Object $post_id       The Post object ID.
```
Comment (juanfra) [inline src/views/v2/tickets/item/extra/price.php:17]: I think this should be `int`.

File: tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/Available/QuantityTest.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	/**
+	 * @test
+	 */
+	public function test_should_not_render_quantity_block_if_threshold_smaller_than_available_quantity() {
+		$template = tribe( 'tickets.editor.template' );
+
+		$args = $this->get_default_args();
+
+		// Threshold is less than the set capacity of 20
```
Comment (juanfra) [inline tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/Available/QuantityTest.php:59]:
```suggestion
		// Threshold is less than the set capacity of 20.
```

File: tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/Available/QuantityTest.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+	/**
+	 * @test
+	 */
+	public function test_should_render_quantity_block_if_threshold_greater_than_available_quantity() {
+		$template = tribe( 'tickets.editor.template' );
+
+		$args = $this->get_default_args();
+
+		// Threshold is greater than the set capacity of 20
```
Comment (juanfra) [inline tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/Available/QuantityTest.php:75]:
```suggestion
		// Threshold is greater than the set capacity of 20.
```

File: tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/Available/UnlimitedTest.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+	/**
+	 * @test
+	 */
+	public function test_should_not_render_if_available_quantity_is_not_unlimited() {
+		$template = tribe( 'tickets.editor.template' );
+
+		$args = $this->get_default_args();
+
+		// set a fixed capacity
```
Comment (juanfra) [inline tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/Available/UnlimitedTest.php:66]:
```suggestion
		// set a fixed capacity.
```

File: tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/PriceTest.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+	/**
+	 * @test
+	 */
+	public function test_should_render_price_block_with_price_suffix() {
+		$template = tribe( 'tickets.editor.template' );
+
+		$args = $this->get_default_args();
+
+		// Set price suffix
```
Comment (juanfra) [inline tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/PriceTest.php:51]:
```suggestion
		// Set price suffix.
```

File: tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/PriceTest.php [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+		$args['ticket']->price_suffix = 'Great_Price';
+
+		$html = $template->template( $this->partial_path, $args, false );
+
+		// Make sure we have suffix block
```
Comment (juanfra) [inline tests/wpunit/Tribe/Tickets/Partials/V2/Tickets/Item/Extra/PriceTest.php:56]:
```suggestion
		// Make sure we have suffix block.
```

**Review by juanfra — CHANGES_REQUESTED** [review]
Great stuff!

File: src/views/v2/tickets/item/extra/available.php [hunk 8]
```diff
@@ -14,8 +14,8 @@
  *
  * @version TBD
  *
- * @var Tribe__Tickets__Editor__Template $this
- * @var Tribe__Tickets__Ticket_Object    $ticket
+ * @var Tribe__Tickets__Editor__Template $this The Template Object
+ * @var Tribe__Tickets__Ticket_Object    $ticket The Ticket Object
```
Comment (sc0ttkclark) [inline src/views/v2/tickets/item/extra/available.php:18]:
```suggestion
 * @var Tribe__Tickets__Editor__Template $this The Template Object.
 * @var Tribe__Tickets__Ticket_Object    $ticket The Ticket Object.
```

File: src/views/v2/tickets/item/extra/available/quantity.php [hunk 9]
```diff
@@ -14,18 +14,17 @@
  *
  * @version TBD
  *
- * @var Tribe__Tickets__Editor__Template $this
- * @var Tribe__Tickets__Ticket_Object    $ticket
- * @var int                              $threshold The threshold.
+ * @var Tribe__Tickets__Editor__Template $this The Template Object
+ * @var Tribe__Tickets__Ticket_Object $ticket  The Ticket Object
+ * @var int $threshold                         The threshold value to show or hide quantity available
```
Comment (sc0ttkclark) [inline src/views/v2/tickets/item/extra/available/quantity.php:19]:
```suggestion
 * @var Tribe__Tickets__Editor__Template $this The Template Object.
 * @var Tribe__Tickets__Ticket_Object $ticket  The Ticket Object.
 * @var int $threshold                         The threshold value to show or hide quantity available.
```

File: src/views/v2/tickets/item/extra/available/unlimited.php [hunk 10]
```diff
@@ -23,7 +23,10 @@
  *
  * @since 4.11.1
  *
- * @param int $show_unlimited allow showing of "unlimited".
+ * @var Tribe__Tickets__Editor__Template $this The Template Object
+ * @var Tribe__Tickets__Ticket_Object $ticket  The Ticket Object
+ * @var int $threshold                         The threshold value to show or hide quantity available
```
Comment (sc0ttkclark) [inline src/views/v2/tickets/item/extra/available/unlimited.php:28]:
```suggestion
 * @var Tribe__Tickets__Editor__Template $this The Template Object.
 * @var Tribe__Tickets__Ticket_Object $ticket  The Ticket Object.
 * @var int $threshold                         The threshold value to show or hide quantity available.
```

File: src/views/v2/tickets/item/extra/available/unlimited.php [hunk 11]
```diff
@@ -23,7 +23,10 @@
...
+ * @var int $show_unlimited                    allow showing of "unlimited".
```
Comment (sc0ttkclark) [inline src/views/v2/tickets/item/extra/available/unlimited.php:29]:
```suggestion
 * @var int $show_unlimited                    Allow showing of "unlimited".
```

File: src/views/v2/tickets/item/extra/price.php (hunk 2 above)
Comment (sc0ttkclark) [reply src/views/v2/tickets/item/extra/price.php:17]:
```suggestion
 * @var int $post_id       The Post object ID.
```

File: src/views/v2/tickets/item/extra/price.php [hunk 12]
```diff
@@ -14,11 +14,11 @@
...
+ * @var Tribe__Tickets__Ticket_Object $ticket        The ticket object.
+ * @var Tribe__Tickets__Tickets $provider            The tickets provider class.
+ * @var string $provider_id                          The tickets provider class name.
+ * @var Tribe__Tickets__Commerce__Currency $currency The Currency Object
```
Comment (sc0ttkclark) [inline src/views/v2/tickets/item/extra/price.php:21]:
```suggestion
 * @var Tribe__Tickets__Commerce__Currency $currency The Currency Object.
```

**Review by sc0ttkclark — CHANGES_REQUESTED** [review]
Some phpdoc tweaks

**Review by sc0ttkclark — APPROVED** [review]

File: src/views/v2/tickets/item/extra/price.php (hunk 12 above)
Comment (sc0ttkclark) [reply src/views/v2/tickets/item/extra/price.php]: @rafsuntaskin be sure to let the reviewer or a lead dev resolve code review comments so we can ensure nothing gets left behind.

**Pull request merged**